File: LA01/qsort.hs
-- qsort.hs

quicksort :: [Int] -> [Int] 
quicksort [] = []
quicksort (x:xs) = quicksort [e | e <- xs, e < x]
                    ++ [x] 
                    ++ quicksort [e | e <- xs, e >= x] 

main :: IO()
main = do
    print (quicksort [2, 6, 0, 9, 2, 1])

File: LA01/qsort.py
# qsort.py

def quicksort(arr):
    if len(arr) <= 1:
        return arr

    pivot = arr[0]

    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]

    return quicksort(left) + middle + quicksort(right)

lst = [2, 6, 0, 9, 2, 1]

print(quicksort(lst))